### PR TITLE
Quick fix for compilation warnings

### PR DESCRIPTION
**Pull request by caioaao**
Just a bunch of comparisons between ints and unsigned ints

### CODE REVIEW

File: sptree.cpp [hunk 1]
```diff
@@ -108,11 +108,11 @@ SPTree<T, dimension>::SPTree(T* inp_data, unsigned int N)
         }
         nD += D;
     }
-    for(int d = 0; d < D; d++) mean_Y[d] /= (T) N;
+    for(int d = 0; d < (int)D; d++) mean_Y[d] /= (T) N;
```
Comment (maximsch2) [inline sptree.cpp:111]: Shouldn't we make `d` to be `unsigned` instead?

Comment (caioaao) [reply sptree.cpp:111]: I don't think It makes any difference. Is there a reason for using unsigneds everywhere? I thought of changing the other ints to unsigneds, but didn't know If it'd affect the code, so just made explicit conversions when needed

Comment (maximsch2) [reply sptree.cpp:111]: In this case, I agree there is no difference in practice, but if we are counting dimensions then it makes sense to have that counter be an unisigned int.

**Comment by maximsch2** [comment]
I've merged it in. Thanks again!

**Pull request merged**